src/hooks/transactions/useTransactionsStatus.tsx

```diff
@@ -0,0 +1,112 @@
+
+import useChainGetters from '../chain/useChainGetters';
+import { useSubscriptionsContext } from '../../contexts/SubscriptionsContextProvider';
+import { getChainSubscriptionsKey } from '../../util/chainsUtils';
+import { handleTransactionUpdates } from '../../util/transactionUtils';
+
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import useLaneId from '../chain/useLaneId';
+import { useEffect } from 'react';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import isEqual from 'lodash/isEqual';
+
+export default function useTransactionsStatus(transactions: TransactionStatusType[], dispatchTransaction: Function) {
+  const { dispatchMessage } = useUpdateMessageContext();
+  const { getValuesByChain } = useChainGetters();
+
+  const subscriptions = useSubscriptionsContext();
+
+  const {
+    sourceChainDetails: { chain: currentSourceChain }
+  } = useSourceTarget();
+  const apiCalls = useApiCallsContext();
+  const laneId = useLaneId();
+
+  useEffect(() => {
+    const getTransactionStatus = async () => {
+      const updatedTransactions = await Promise.all(
+        transactions.map(async (transaction: TransactionStatusType) => {
+          if (transaction.status === TransactionStatusEnum.COMPLETED || transaction.evaluating) {
+            return transaction;
+          }
+          dispatchTransaction(
+            TransactionActionCreators.updateTransactionStatus(
+              {
+                evaluating: true
+              },
+              transaction.id
+            )
+          );
+          const { sourceChain, targetChain } = transaction;
+          const { api: targetApi } = getValuesByChain(targetChain);
+          const { sourceRole, targetRole } = getChainSubscriptionsKey({
+            currentSourceChain,
+            sourceChain
+          });
+
+          const sourceSubscriptions = subscriptions[sourceRole];
+          const targetSubscriptions = subscriptions[targetRole];
+
+          const updatedTransaction = await handleTransactionUpdates({
+            transaction,
+            sourceSubscriptions,
+            targetSubscriptions,
+            apiCalls: {
+              targetApi,
+              ...apiCalls
+            },
+            laneId,
+            dispatchMessage
+          });
+
+          if (isEqual(transaction, updatedTransaction)) {
+            dispatchTransaction(
+              TransactionActionCreators.updateTransactionStatus(
+                {
+                  evaluating: false
+                },
+                transaction.id
+              )
+            );
+          }
+
+          return updatedTransaction;
+        })
+      );
+
+      if (!isEqual(transactions, updatedTransactions)) {
+        dispatchTransaction(TransactionActionCreators.updateTransactionsStatus(updatedTransactions));
+      }
+    };
+
+    getTransactionStatus();
```

Comment:
async errors are being swallowed here + you risk dispatching actions after the hook is unmounted (since it's in the context, I guess that's not a real issue, but we could add a comment somewhere).

Reply:
I tried to address both concerns:

1) In case there is any async error, an error message will be dispatched so the toast appears on the screen.
2) If due to really edged case the component gets unmounted while the evaluation is running, i make sure to set the transaction evaluation state to false again so the next cycle can re start the process.